src/blocks/scratch3_looks.js

```diff
@@ -409,7 +409,17 @@ class Scratch3LooksBlocks {
         const instance = this;
         const waiting = util.stackFrame.startedThreads.some(thread => instance.runtime.isActiveThread(thread));
         if (waiting) {
-            util.yield();
+            // If all threads are waiting for the next tick or later yield
+            // for a tick as well. Otherwise yield until the next loop of
+            // the threads.
+            if (
+                util.stackFrame.startedThreads
+                    .some(thread => !instance.runtime.isWaitingThread(thread)) === false
```

Comment:
I can never fully understand double negatives no matter how many time I read them. Is this any different than

```js
               util.stackFrame.startedThreads
                   .every(thread => instance.runtime.isWaitingThread(thread))
```
? And if not can we change it to that so I can put my brain back right side in?

Reply:
Er..  I guess the opposite of what I said. Not every one is waiting?
```js
               !util.stackFrame.startedThreads
                   .every(thread => instance.runtime.isWaitingThread(thread))
```

Reply:
This is one way to prove your point. :wink:

Your first guess was right. Everyone is waiting.

Reply:
I promise I didn't get confused on purpose! Thanks for changing that.